pkg/kubeapiserver/admission/util/initializer.go

```diff
@@ -49,8 +50,45 @@ func IsUpdatingUninitializedObject(a admission.Attributes) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	if accessor.GetInitializers() == nil {
+	if initialization.IsInitialized(accessor.GetInitializers()) {
 		return false, nil
 	}
 	return true, nil
 }
+
+// IsInitializationCompletion returns if the operation removes all pending initializers.
+func IsInitializationCompletion(a admission.Attributes) (bool, error) {
```

Comment:
if the initializer feature is disabled, this should always return false, right?

Reply:
similar short-circuits for the other helpers, though the package dependency for that is pretty bad, I guess :-/

Reply:
The old object is persisted. If the feature gate is off, old object will always be initialized, so all these helpers (IsUpdatingInitializedObject, IsUpdatingUninitializedObject, IsInitializationCompletion) will return false. No need to protect them with feature gate.